tests/protocol/test_ipc.py

```diff
@@ -102,3 +107,47 @@ def get_context(self) -> None:
     args = [1, 2, 3, 4]
     result = await client_protocol.call(Echo(args))
     assert args == result
+
+
+async def test_log_transport(caplog, request):
+    """
+    Test for the IPC feature of shipping logs
+
+    Ensure we can send a log line over the IPC protocol
+
+    This doesn't actually connect the LogShipper to the logging framework, as this would create an infinite loop
+    (the log is re-injected in the same place it was first received)
+    """
+
+    loop = asyncio.get_running_loop()
+    parent_conn, child_conn = socket.socketpair()
+
+    server_transport, server_protocol = await loop.connect_accepted_socket(
+        lambda: inmanta.protocol.ipc_light.LogReceiver("logs"), parent_conn
+    )
+    request.addfinalizer(server_transport.close)
+    client_transport, client_protocol = await loop.connect_accepted_socket(lambda: IPCClient("Client"), child_conn)
+    request.addfinalizer(client_transport.close)
+    log_shipper = inmanta.protocol.ipc_light.LogShipper(client_protocol, loop)
```

Comment:
I am quite confused here:

- you have a transport and a protocol
- but the logshipper has an argument channel to which you pass the client. However, based on naming I would have assumed that the transport is the channel?

Reply:
updated